Sources/TuistKit/Graph/GraphManifestLoader.swift

```diff
@@ -138,23 +128,41 @@ class GraphManifestLoader: GraphManifestLoading {
         })
     }
 
+    func loadProject(at path: AbsolutePath) throws -> ProjectDescription.Project {
+        return try loadManifest(.project, at: path)
+    }
+    
+    func loadWorkspace(at path: AbsolutePath) throws -> ProjectDescription.Workspace {
+        return try loadManifest(.workspace, at: path)
+    }
+    
     func loadSetup(at path: AbsolutePath) throws -> [Upping] {
-        let setupPath = path.appending(component: "Setup.swift")
+        let setupPath = path.appending(component: Manifest.setup.fileName)
         guard fileHandler.exists(setupPath) else {
-            throw GraphManifestLoaderError.setupNotFound(path)
+            throw GraphManifestLoaderError.manifestNotFound(.setup, path)
         }
 
-        let setup = try loadManifest(path: setupPath)
-        let actionsJson: [JSON] = try setup.get("actions")
+        let setup = try loadManifestData(at: setupPath)
+        let setupJson = try JSON(data: setup)
+        let actionsJson: [JSON] = try setupJson.get("actions")
```

Comment:
This is the last remaining `JSON` type is - we could try out using the `ProjectDescription` type directly seeing they are 1-1. Code footprint will still be similar as we'll need to manually decode the individual actions (they currently subclass `Up`) - an alternative is transforming `Up` into an enum and leverage `AutoCodable` from sourcery.

Reply:
Another idea. Given that having those `Up` subclasses of `Up` doesn't make much sense to me, I think we could use a more protocol based approach:

```swift
protocol Up: Codable {}
extension Up {
  public static func homebrew(packages: [String]) -> Up {
    // initialize homebrew up
  }
}
class UpHomebrew: Up {
}

```
Not sure at all if this is feasible *(it's been a while without playing a lot with Swift)*